src/main/java/org/openstreetmap/atlas/streaming/resource/InputStreamResource.java

```diff
@@ -16,11 +16,28 @@
 {
     private final Supplier<InputStream> inputStreamSupplier;
 
+    /**
+     * This constructor is deprecated. Creating an InputStreamResource using a raw InputStream leads
+     * to many gotchas, since the resource becomes read-once. For e.g. loading an atlas from an
+     * InputStreamResource will fail, since atlas uses an under-the-hood read to determine the
+     * serialization format.
+     *
```

Comment:
This needs to have `@deprecated` in the javadoc too. [See sonar](
https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas&issues=AWjI-m_sChQW8w82rtlG&open=AWjI-m_sChQW8w82rtlG&pullRequest=339).

Reply:
Ah ok. The Eclipse plugin did not catch this.